src/main/java/com/artipie/api/RepositoryRest.java

```diff
@@ -0,0 +1,182 @@
+        rbr.operation("listAll")
+            .handler(this::listAll)
+            .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+        if ("flat".equals(this.layout)) {
+            rbr.operation("getRepo")
+                .handler(this::getRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+            rbr.operation("createRepo")
+                .handler(this::createRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+        } else {
+            rbr.operation("list")
+                .handler(this::listUserRepos)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+            rbr.operation("getUserRepo")
+                .handler(this::getRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+            rbr.operation("createUserRepo")
+                .handler(this::createRepo)
+                .failureHandler(this.errorHandler(HttpStatus.INTERNAL_SERVER_ERROR_500));
+        }
+        final Router router = rbr.createRouter();
+        router.route("/api/*").handler(
+            StaticHandler.create(
+                FileSystemAccess.ROOT,
+                new JavaResource("swagger-ui").uri().getPath()
+            )
+        );
+    }
+
+    /**
+     * Get a repository settings json.
+     * @param context Routing context
+     */
+    private void getRepo(final RoutingContext context) {
+        final RepositoryName rname = new RepositoryName.FromRequest(context, this.layout);
+        if (!this.crs.exists(rname)) {
+            context.response()
+                .setStatusCode(HttpStatus.NOT_FOUND_404)
+                .end(String.format("Repository %s does not exist", rname));
+            return;
+        }
+        context.response().setStatusCode(HttpStatus.OK_200).end(
+            this.crs.value(rname).toString()
+        );
+    }
+
+    /**
+     * List all existing repositories.
+     * @param context Routing context
+     */
+    private void listAll(final RoutingContext context) {
+        context.response().setStatusCode(HttpStatus.OK_200).end(
+            JsonArray.of(this.crs.listAll().toArray()).encode()
+        );
+    }
+
+    /**
+     * List user's repositories.
+     * @param context Routing context
+     */
+    private void listUserRepos(final RoutingContext context) {
+        final String uname = context.pathParam(RepositoryName.UNAME);
+        context.response().setStatusCode(HttpStatus.OK_200).end(
+            JsonArray.of(this.crs.list(uname).toArray()).encode()
+        );
+    }
+
+    /**
+     * Create a repository.
+     * @param context Routing context
+     */
+    private void createRepo(final RoutingContext context) {
+        final RepositoryName rname = new RepositoryName.FromRequest(context, this.layout);
+        if (this.crs.exists(rname)) {
```

Comment:
If repo name is a service name ("_storages", "_permissions", "_credentials") then this method returns false, 
and we will create repo with a service name.
It's wrong.